posthog/schema_migrations/base.py

```diff
@@ -0,0 +1,31 @@
+from abc import ABC, abstractmethod
+
+from posthog.schema import NodeKind
+
+
+class SchemaMigration(ABC):
+    """
+    Base class for schema migrations. Bumps the schema of one or more query types by one version.
+
+    Subclasses must define:
+    * `targets`: A mapping of node kinds to the version before running the migration.
```

Comment:
each node kind has a different version?
just curious if there's some big disadvantages over having a single version for the entire query?

Reply:
yes. that allows us to selectively upgrade only insights that contain a node which has changed. means less work to do for every migration (the filters to query migration took many hours) and less cache invalidations.

Reply:
Great point, this is very neat!